runtimeconfig/manager_test.go

```diff
@@ -54,26 +55,48 @@ func testLoadOverrides(r io.Reader) (interface{}, error) {
 	return overrides, nil
 }
 
-func newTestOverridesManagerConfig(t *testing.T, i int32) (*atomic.Int32, Config) {
-	var config = atomic.NewInt32(i)
+type value struct {
+	Value int `yaml:"value"`
+}
+
+func valueLoader(r io.Reader) (i interface{}, err error) {
+	v := value{Value: 0}
+	buf, err := io.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
 
+	err = yaml.Unmarshal(buf, &v)
+	if err != nil {
+		return nil, err
+	}
+
+	return v, nil
+}
+
+func writeValueToFile(t *testing.T, path string, v value) {
+	buf, err := yaml.Marshal(v)
```

Comment:
[Nit] I think test utility functions should be marked with `t.Helper()`.
```suggestion
	t.Helper()

	buf, err := yaml.Marshal(v)
```